test/common/upstream/edf_scheduler_test.cc

```diff
@@ -1,19 +1,55 @@
 #include "source/common/upstream/edf_scheduler.h"
 
+#include "test/test_common/test_random_generator.h"
+
 #include "gtest/gtest.h"
 
 namespace Envoy {
 namespace Upstream {
-namespace {
 
-TEST(EdfSchedulerTest, Empty) {
+class EdfSchedulerTest : public testing::Test {
+public:
+  template <typename T>
+  static void compareEdfScehdulers(EdfScheduler<T>& scheduler1, EdfScheduler<T>& scheduler2) {
+    // Compares that the given EdfSchedulers internal queues are equal up
+    // (ignoring the order_offset_ values).
+    EXPECT_EQ(scheduler1.queue_.size(), scheduler2.queue_.size());
+    // Cannot iterate over std::priority_queue directly, so need to copy the
+    // contents to a vector first.
+    auto copyFunc = [](EdfScheduler<T>& scheduler) {
+      std::vector<typename EdfScheduler<T>::EdfEntry> result;
+      result.reserve(scheduler.queue_.size());
+      while (!scheduler.empty()) {
+        result.emplace_back(std::move(scheduler.queue_.top()));
+        scheduler.queue_.pop();
+      }
+      // Re-add all elements so the contents of the input scheduler isn't
+      // changed.
+      for (auto& entry : result) {
+        scheduler.queue_.push(entry);
+      }
+      return result;
+    };
+    std::vector<typename EdfScheduler<T>::EdfEntry> contents1 = copyFunc(scheduler1);
+    std::vector<typename EdfScheduler<T>::EdfEntry> contents2 = copyFunc(scheduler2);
+    for (size_t i = 0; i < contents1.size(); ++i) {
+      EXPECT_NEAR(contents1[i].deadline_, contents2[i].deadline_, 1e-5)
```

Comment:
nit: can we add explanatory comment why do expect deadlines to be equal within absolute error bound `1e-5`?

Reply:
Done.